Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -803,7 +805,21 @@ static void registerLogHook()
         if (type == OS_LOG_TYPE_DEBUG)
             return;
 
-        CString logFormat(msg->format);
+        os_log_message_s msgStr;
+        memset(&msgStr, 0, sizeof(msgStr));
+
+        msgStr.format = msg->format;
+        msgStr.buffer = msg->buffer;
+        msgStr.buffer_sz = msg->buffer_sz;
+        msgStr.privdata = msg->privdata;
+        msgStr.privdata_sz = msg->privdata_sz;
+        msgStr.subsystem = msg->subsystem;
+        msgStr.category = msg->category;
+
+        UniqueMessageBuffer messageString { os_log_copy_message_string(&msgStr), std::free };
```

Comment:
Doesn't it defeat some of the purpose of using a work queue? My understanding is that Per intentionally was doing the string formatting off the main thread and it now occurs on the main thread.